.circleci/workflows.yml

```diff
@@ -488,6 +510,61 @@ commands:
             which google-chrome-<<parameters.channel>> || (printf "\n\033[0;31mChrome was not successfully downloaded - bailing\033[0m\n\n" && exit 1)
             echo "Location of Google Chrome Installation: `which google-chrome-<<parameters.channel>>`"
             echo "Google Chrome Version: `google-chrome-<<parameters.channel>> --version`"
+  
+  # This code builds better-sqlite3 on CentOS 7. This is necessary because CentOS 7 has the oldest glibc version
+  # that we support. The script checks for the existence of the Centos7-builder image tar file, and skips if it already
+  # exists. If you want to rebuild the image, set the REBUILD_CENTOS_BUILDER_IMAGE environment variable to any value.
+  # Since this is running Docker remote, we need to copy the project into the container, and copy the built plugin out
+  # of the container because the host running docker does not have access to the project directory so volume mounts are
+  # not possible. The built plugin is copied to the project directory so it can be injected into the final binary.  
+  build-better-sqlite3:
+    description: Build better-sqlite3 for CentOS 7
+    steps:
+      - setup_remote_docker
+      - restore_cache:
+          keys:
+            - cypress-centos7-builder-{{ checksum "centos7-builder.Dockerfile" }}
+      - run:
```

Comment:
Seems like we could make a slight optimization here... We're restoring this cache and loading the Docker image (which is taking like almost 1 minute) here every time, even when we're not going to re-build better-sqlite.

I wonder if we could do these steps conditionally based on whether `better_sqlite3.node` is restored from cache? I've had trouble doing this type of thing in the past, so I'm not sure it's possible.

One thing that I think we might be able to do is store `centos7-builder.tar` and `better_sqlite3.node` under the same cache key, since we know that we'll only need the Docker image if we need to re-build `better_sqlite3.node`. Then, restore the cache and if the file doesn't exist, load the image in and re-build better_sqlite3. 

Just something to look into. Let me know if you want to talk it out over a call. Would be nice to make this as efficient as possible since our CI is already super long

Reply:
I think I found a good way to do this. Thoughts?